### PR TITLE
Issue 152 - adding support for XEC - Remove sats and bits as currencies

### PR DESCRIPTION
**Pull request by Gattaca1**
BCH functionality should remain intact - but `BITS` and `SATS` have been removed as currency types

### CODE REVIEW

**Review by Klakurka — CHANGES_REQUESTED** [review]
I know that this was part of a previous branch that just wasn't caught / tested but we should still fix it:

While up: 
image

While animating away:

image

**Review by Klakurka — APPROVED** [review]
Ok I'm going to approve this because the issue is going to get fixed in the next branch.

File: react/src/components/Widget/Widget.tsx [hunk 1]
```diff
@@ -372,29 +364,18 @@ export const Widget: React.FC<WidgetProps> = props => {
   };
 
   useEffect(() => {
-    const inSatoshis = getCurrencyObject(totalSatsReceived, 'SAT');
+    // const inSatoshis = getCurrencyObject(totalSatsReceived, 'SAT');
```
Comment (ScottMcDermid) [inline react/src/components/Widget/Widget.tsx:367]: Do we need to keep this?

File: react/src/components/Widget/Widget.tsx [hunk 2]
```diff
@@ -186,11 +183,7 @@ export const Widget: React.FC<WidgetProps> = props => {
   const query: string[] = [];
   const isMissingWidgetContainer = !totalReceived;
   const addressDetails = useAddressDetails(to, isMissingWidgetContainer);
-  const isFiat: boolean =
-    currency !== 'SAT' &&
-    currency !== 'BCH' &&
-    currency !== 'bits' &&
-    currency !== 'XEC';
+  const isFiat: boolean = currency !== 'BCH' && currency !== 'XEC';
```
Comment (ScottMcDermid) [inline react/src/components/Widget/Widget.tsx:186]: `isFiat` seems like a core function. Might deserve to be in a currency service.
Also I'd expect this to look more like `fiatCurrencies.includes(currency)`. Especially if we end up adding another cryptocurrency.

**Review by ScottMcDermid — CHANGES_REQUESTED** [review]

File: react/src/components/Widget/Widget.tsx (hunk 2 above)
Comment (Gattaca1) [reply react/src/components/Widget/Widget.tsx:186]: I'm not quite sure how to do `fiatCurrencies.includes(currency)` the typescript way yet. The possible currencies themselves are types. Hm. Maybe by importing `isFiat` and type checking it from the utils folder. Alright I have some ideas for this

File: react/src/components/Widget/Widget.tsx (hunk 1 above)
Comment (Gattaca1) [reply react/src/components/Widget/Widget.tsx:367]: I wont tell anyone if you don't

(I fixed it in the next stacked branch, I'll see if i can do it here without fucking up git)

File: react/src/components/Widget/Widget.tsx (hunk 2 above)
Comment (Gattaca1) [reply react/src/components/Widget/Widget.tsx:186]: In the next stacked branch I moved `isFiat` to `util/api-client.ts` which is functionally serving as a currency service

**Comment by Gattaca1** [comment]
> I know that this was part of a previous branch that just wasn't caught / tested but we should still fix it:

Lets bump existing issues / issues that affect both BCH and XEC to their own branch. Trying to keep the scope of this branch purely to adding XEC support

**Review by ScottMcDermid — APPROVED** [review]

**Pull request merged**